client/components/filter-select-control/index.tsx

```diff
@@ -0,0 +1,250 @@
+/**
+ * This is a copy of Gutenberg's CustomSelectControl component, found here:
+ * https://github.com/WordPress/gutenberg/tree/7aa042605ff42bb437e650c39132c0aa8eb4ef95/packages/components/src/custom-select-control
+ *
+ * It has been forked from the existing WooPayments copy of this component (client/components/custom-select-control)
+ * to match this specific select input design with an inline label and option hints.
+ */
+
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { Button } from '@wordpress/components';
+import { check, chevronDown, Icon } from '@wordpress/icons';
+import { useCallback } from '@wordpress/element';
+import classNames from 'classnames';
+import { __, sprintf } from '@wordpress/i18n';
+import { useSelect, UseSelectState } from 'downshift';
+
+/**
+ * Internal Dependencies
+ */
+import './style.scss';
+
+export interface Item {
```

Comment:
`Item` is pretty amorphous, can we give this a clearer name? I like `DropdownItem`, but that's because I'm familiar with macOS UI patterns/naming.

[Looks like the current naming for this component in mac is `Pop-up button`](https://developer.apple.com/design/human-interface-guidelines/pop-up-buttons). So if we follow that the items would be `PopupButtonItem`s. Or if we stick with `Select`, `SelectItem`s, or `SelectControlItem`s.

Reply:
The name `item` was used by the Gutenberg `CustomSelectControl`. I assume this naming was used to match the [`downshift` library API and naming](https://github.com/downshift-js/downshift/tree/master/src/hooks/useSelect#basic-props).

https://github.com/Automattic/woocommerce-payments/blob/b56b86b1a5fda78cfc57f18b674d2d93ed621c89/client/components/filter-select-control/index.tsx/#L127-L134

However, there is no harm in changing the name of the TS `Item` interface to be more descriptive. I will make this change.